instrumentation/src/androidTest/java/com/bumptech/glide/DarkModeTest.java

```diff
@@ -51,6 +52,45 @@ public void before() {
     assumeTrue(VERSION.SDK_INT >= VERSION_CODES.Q);
   }
 
+  @Test
+  public void load_withDarkModeActivity_vectorDrawable_usesDarkModeColor() {
+    try (ActivityScenario<FragmentActivity> scenario = darkModeActivity()) {
+      scenario.onActivity(
+          activity -> {
+            ViewGroup container = findContainer(activity);
+            ImageView imageView = newFixedSizeImageView(activity);
+            container.addView(imageView);
+
+            Glide.with(activity)
+                .load(R.drawable.vector_drawable)
+                .override(Target.SIZE_ORIGINAL)
+                .into(imageView);
+          });
+
+      onIdle();
+      scenario.onActivity(
+          activity -> {
+            ViewGroup container = findContainer(activity);
+            ImageView imageView = (ImageView) container.getChildAt(0);
+            Bitmap result = drawToBitmap(imageView.getDrawable());
+            Bitmap expected = drawToBitmap(activity.getDrawable(R.drawable.vector_drawable_dark));
+            assertThat(result).sameAs(expected);
+          });
+    }
+  }
+
+  private static Bitmap drawToBitmap(Drawable drawable) {
+    int width = drawable.getIntrinsicWidth();
+    int height = drawable.getIntrinsicHeight();
+
+    Bitmap result = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+    Canvas canvas = new Canvas(result);
+    drawable.setBounds(0, 0, width, height);
+    drawable.draw(canvas);
+    canvas.setBitmap(null);
+    return result;
+  }
```

Comment:
**Suggestion**: Consider breaking down the test method load_withDarkModeActivity_vectorDrawable_usesDarkModeColor into smaller, more focused test cases. This can improve test readability and make it easier to identify specific points of failure. <br> **Code Suggestion**: 
 ```
 + // Suggestion to split into smaller tests if applicable
 + @Test public void loadVectorDrawable_inDarkModeActivity_checksDarkModeColor() {
 ```